targets/TARGET_STM/us_ticker_32b.c

```diff
@@ -48,6 +48,10 @@ void us_ticker_set_interrupt(timestamp_t timestamp)
     __HAL_TIM_SET_COMPARE(&TimMasterHandle, TIM_CHANNEL_1, (uint32_t)timestamp);
     // Enable IT
     __HAL_TIM_ENABLE_IT(&TimMasterHandle, TIM_IT_CC1);
+    // Check if timestamp has already passed, and if so, call the IRQ handler immediately
+    if ((int32_t)(timestamp - TIM_MST->CNT) <= 0) {
```

Comment:
I'd rather move this code before the SET_COMPARE and ENABLE_IT calls.
If you're in this situation you can call the handler then return - don't you think ?